Source/i_sound.c

```diff
@@ -200,51 +201,51 @@ static boolean addsfx(sfxinfo_t *sfx, int channel, int pitch)
       // [crispy] Check if this is a valid RIFF wav file
       if (lumplen > 44 && memcmp(data, "RIFF", 4) == 0 && memcmp(data + 8, "WAVEfmt ", 8) == 0)
       {
-         // Valid RIFF wav file
-         int check;
+        SDL_RWops *RWops;
+        SDL_AudioSpec wav_spec;
 
-         // Make sure this is a PCM format file
-         // "fmt " chunk size must == 16
-         check = data[16] | (data[17] << 8) | (data[18] << 16) | (data[19] << 24);
-         if (check != 16)
-         {
-            Z_ChangeTag(data, PU_CACHE);
-            return false;
-         }
+        RWops = SDL_RWFromMem(data, lumplen);
 
-         // Format must == 1 (PCM)
-         check = data[20] | (data[21] << 8);
-         if (check != 1)
-         {
-            Z_ChangeTag(data, PU_CACHE);
-            return false;
-         }
+        Z_ChangeTag(data, PU_CACHE);
 
-         // FIXME: can't handle stereo wavs
-         // Number of channels must == 1
-         check = data[22] | (data[23] << 8);
-         if (check != 1)
-         {
-            Z_ChangeTag(data, PU_CACHE);
+        if (SDL_LoadWAV_RW(RWops, 1, &wav_spec, &wav_buffer, &samplelen) == NULL)
```

Comment:
I've read a bit about the WAV format and it's not that simple. I think it's better to use SDL here.

Reply:
Even better!